Xamarin.Forms.Core/ListView.cs

```diff
@@ -271,11 +271,11 @@ public Element HeaderElement
 
 		[EditorBrowsable(EditorBrowsableState.Never)]
 		public void SendCellAppearing(Cell cell)
-			=> ItemAppearing?.Invoke(this, new ItemVisibilityEventArgs(cell.BindingContext));
+			=> ItemAppearing?.Invoke(this, new ItemVisibilityEventArgs(cell.BindingContext, TemplatedItems.GetGlobalIndexOfItem(cell?.BindingContext)));
```

Comment:
what's the cost of GetIndex ?

Reply:
please figure that out. if the complexity is too high, we'll have to close this PR

Reply:
When not grouped it should be O(N), it goes to a proxy object and just gets the index.

    public int GetGlobalIndexOfItem(object item)
    {
	if (!IsGroupingEnabled)
		return ListProxy.IndexOf(item);
        ....
    }

When grouped, it loops once for sure, but shouldn't be different from the `IndexOf` and then loops again recursively. Making it O(N^2).

    if (_groupedItems != null)
    {
        foreach (TemplatedItemsList<TView, TItem> children in _groupedItems.Values)
        {
	        count++;

	        int index = children.GetGlobalIndexOfItem(item);
	        if (index != -1)
	            return count + index;

	        count += children.GetDescendantCount();
        }
    }

I also went ahead and added a stopwatch and tested with appearing, dissapearing and selecting which are all sub 1ms or even sub .1ms. Full version: https://www.dropbox.com/s/hfnvtdvn8mt96br/StopwatchDemo.mov?dl=0

![Stopwatch test](https://user-images.githubusercontent.com/939291/48697054-e8c30a80-ebe3-11e8-930d-3c13e041ff8e.gif)

But maybe I'm overlooking something?

Reply:
is it worth trying to expose this as `Lazy`? @hartez, as you've spent quite a lot of time with LV lately, could you weight in one way or the other ?

Reply:
> is it worth trying to expose this as `Lazy`? @hartez, as you've spent quite a lot of time with LV lately, could you weight in one way or the other ?

@StephaneDelcroix 

Which property are you proposing to make `Lazy`? Or are you asking if we should cache the index in the `Cell`?

Reply:
@jfversluis Have you checked the performance characteristics on Android? That platform would be much more of a concern.

Reply:
Valid point. Tried the same on Android: https://www.dropbox.com/s/vdafery4lulgeat/StopwatchDemoAndroid.mov?dl=0

Times are a bit higher but overall no more than 3ms with some exceptions going to 26ms (end of video). It's a bit hard to see because the bindings seem to be broken on Android for some reason (as far as I can tell not related to this change). Also, I didn't do anything to enable fast renderers. Not sure if they are enabled for the gallery app or if it should have any impact at all.

Reply:
You can enable the fast renderers in the Control Gallery by building it with the TEST_EXPERIMENTAL_RENDERERS symbol defined. But I don't think they should have any impact on the time spent finding indexes.

Reply:
Tried, doesn't have impact indeed. As far as I'm concerned it's totally up to you guys. I can see how the performance is a crucial thing for the ListView and I don't think this "feature" doesn't outweigh any performance issues that this might cause. At least now we know :)